shell/platform/android/test/io/flutter/embedding/android/FlutterActivityTest.java

```diff
@@ -196,6 +202,23 @@ public void itCanBeDetachedFromTheEngineAndStopSendingFurtherEvents() {
     verify(mockDelegate, times(1)).onDetach();
   }
 
+  @Test
+  public void itRestoresPluginStateBeforePluginOnCreate() {
+    FlutterLoader mockFlutterLoader = mock(FlutterLoader.class);
+    FlutterJNI mockFlutterJni = mock(FlutterJNI.class);
+    when(mockFlutterJni.isAttached()).thenReturn(true);
+    FlutterEngine cachedEngine =
+        new FlutterEngine(RuntimeEnvironment.application, mockFlutterLoader, mockFlutterJni);
+    FakeFlutterPlugin fakeFlutterPlugin = new FakeFlutterPlugin();
+    cachedEngine.getPlugins().add(fakeFlutterPlugin);
+    FlutterEngineCache.getInstance().put("my_cached_engine", cachedEngine);
+
+    Intent intent =
+        FlutterActivity.withCachedEngine("my_cached_engine").build(RuntimeEnvironment.application);
+    Robolectric.buildActivity(FlutterActivity.class, intent).create();
```

Comment:
In the above cases, we're trying to test the activity at various stages of the creation. But if you're just fully launching an activity through the normal sequence, should we just use Robolectric.setupActivity to adhere to the framework's sequence?

Reply:
I went to do that, but that method only accepts a Class object. Since we need the intent, we can't use it.

Reply:
I think we can still use buildActivity and then chain setup() instead of create() to let it retrace a realistic set of lifecycle events. Otherwise, we have to call create, start, postCreate etc in the right sequence ourselves.

Reply:
Done